tests/unit/bokeh/test_resources.py

```diff
@@ -66,11 +66,18 @@ def test_get_sri_hashes_for_version(self) -> None:
         versions = resources.get_all_sri_versions()
         for v in versions:
             h = resources.get_sri_hashes_for_version(v)
-            assert f"bokeh-{v}.js" in h
-            assert f"bokeh-{v}.min.js" in h
-            if not v.startswith("0"):
-                assert f"bokeh-widgets-{v}.js" in h
-                assert f"bokeh-widgets-{v}.min.js" in h
+            if "0.3" <= v:
+                if v == "0.3.0":
+                    v_t = "0.3"
+                elif v == "0.4.0":
+                    v_t = "0.4"
+                else:
+                    v_t = v
+                assert f"bokeh-{v_t}.js" in h
+                assert f"bokeh-{v_t}.min.js" in h
+            if "1" <= v:
+                assert f"bokeh-widgets-{v_t}.js" in h
+                assert f"bokeh-widgets-{v_t}.min.js" in h
```

Comment:
Hi @mosc9575 I think this kind of logic indicates a test as grown over-complicated. There are a couple of things we can do, including splitting the "standard" and "weird" cases into their own tests, and using `parametrize` to avoid an explicit loop level. Given the arbitrary historical nature of these requirements, some comments are also probably helpful for posterity. 

What do you think of an approach like this (untested):

```python
from packaging.version import Version as V

# very old Bokeh versions are inconsistent have have to be handled specially
STANDARD_VERSIONS = {v for v in resources.get_all_sri_versions() if V(v) >= V("0.4")}
WIERD_VERSIONS = {v for v in resources.get_all_sri_versions() if V(v) < V("0.4")}

class TestSRIHashes:
   
   @pytest.mark.parametrize("v", STANDARD_VERSIONS)
    def test_get_sri_hashes_for_standard_versions(self, v) -> None:
        h = resources.get_sri_hashes_for_version(v)
        assert f"bokeh-{v}.js" in h
        assert f"bokeh-{v}.min.js" in h
        if v >= "1":
            assert f"bokeh-widgets-{v}.js" in h
            assert f"bokeh-widgets-{v}.min.js" in h

    @pytest.mark.parametrize("v", WIERD_VERSIONS)
    def test_get_sri_hashes_for_weird_versions(self) -> None:

        # 0.1.0 is untestable 
        if v == "0.1.0": 
            return 
        
        # other early versions omitted trailing .0 in filnames
        v = v.rstrip(".0")

        assert f"bokeh-{v}.js" in h
        assert f"bokeh-{v}.min.js" in h
```

Reply:
@bokeh/core there's also repo tags for `0.0.1` and `0.1.1` that don't have JSON files or releases on PyPI. I would not be opposed to deleting all the pre-0.2 tags (`0.0.1`, `0.1.0` and `0.1.1`) TBH, since I am not sure what useful purpose they serve at this point. That would simplify things here slightly as well.

Reply:
I don't see any purpose in keeping any pre-0.2 tags either. Especially since it sounds like it would make resolving this issue easier, I'd say go for it!

Reply:
Thinking more I suppose we should keep 0.1.0 which is fine, but I will plan to delete the others soon unless there is objection. Then the only code changes for this PR are what I suggested above.